### PR TITLE
arm/efm32: Fix typo error: CONFIG_EFM32_I2C_DYNTIMEOUT to CONFIG_EFM32_I2C_DYNTIMEO

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
Found in #6226 

## Impact
None

## Testing
Pass CI

### CODE REVIEW

File: arch/arm/src/efm32/efm32_i2c.c [hunk 1]
```diff
@@ -513,7 +513,7 @@ static useconds_t efm32_i2c_tousecs(int msgc, struct i2c_msg_s *msgs)
    * factor.
    */
 
-  return (useconds_t) (CONFIG_EFM32_I2C_DYNTIMEO_USECPERBYTE * bytecount);
+  return (useconds_t) (CONFIG_EFM32_I2C_DYNTIMEOUT_USECPERBYTE * bytecount);
```
Comment (pkarashchenko) [inline arch/arm/src/efm32/efm32_i2c.c:516]: Oah... I didn't expect so radical changes.
I think thinking only about changing
```
#ifdef CONFIG_EFM32_I2C_DYNTIMEOUT
static useconds_t efm32_i2c_tousecs(int msgc, struct i2c_msg_s *msgs);
#endif /* CONFIG_EFM32_I2C_DYNTIMEOUT */
```
to 
```
#ifdef CONFIG_EFM32_I2C_DYNTIMEO
static useconds_t efm32_i2c_tousecs(int msgc, struct i2c_msg_s *msgs);
#endif /* CONFIG_EFM32_I2C_DYNTIMEO */
```

Comment (pkarashchenko) [reply arch/arm/src/efm32/efm32_i2c.c:516]: I mean I'm fine with `I2C_DYNTIMEO` (and not changing to `I2C_DYNTIMEOUT`), just would like to have same macro used in all places

Comment (xiaoxiang781216) [reply arch/arm/src/efm32/efm32_i2c.c]: Ok, change CONFIG_EFM32_I2C_DYNTIMEOUT to CONFIG_EFM32_I2C_DYNTIMEO instead.

**Review by hartmannathan — APPROVED** [review]
LGTM

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**